File: backend-java/PXLFullstackJavaProjectBackend/Post-Service/src/main/java/be/pxl/domain/Author.java
package be.pxl.domain;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.*;


@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "authors")
@Entity
public class Author {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @NotBlank(message = "First name is required.")
    @Size(min = 2, max = 50, message = "First name must be between 2 and 50 characters.")
    @Pattern(regexp = "^[a-zA-Z\\s-']+$",
            message = "First name can only contain letters, spaces, hyphens, and apostrophes")
    @Column(name = "first_name")
    private String firstName;

    @NotBlank(message = "Last name is required.")
    @Size(min = 2, max = 50, message = "Last name must be between 2 and 50 characters.")
    @Pattern(regexp = "^[a-zA-Z\\s-']+$",
            message = "Last name can only contain letters, spaces, hyphens, and apostrophes")
    @Column(name = "last_name")
    private String lastName;
}

File: frontend-web/PXLFullstackJavaProjectFrontend/src/app/models/author.ts
export class Author {
  private _id: number;
  private _firstName: string;
  private _lastName: string;


  constructor(id: number, firstName: string, lastName: string) {
    this._id = id;
    this._firstName = firstName;
    this._lastName = lastName;
  }

  toString(): string {
    return this.firstName + " " + this.lastName;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get firstName(): string {
    return this._firstName;
  }

  set firstName(value: string) {
    this._firstName = value;
  }

  get lastName(): string {
    return this._lastName;
  }

  set lastName(value: string) {
    this._lastName = value;
  }
}
